server/idapiProxy.ts

```diff
@@ -0,0 +1,180 @@
+import type { NextFunction, Request, Response } from 'express';
+import type { HTTPMethod } from '../shared/apiTypes';
+import { conf as mmaConfig } from './config';
+import type { IdapiConfig } from './idapiConfig';
+import { getConfig } from './idapiConfig';
+import { handleError } from './util';
+
+export type NewsletterPatchRequest = {
+	id: string;
+	subscribed: boolean;
+};
+
+export type ConsentPatchRequest = NewsletterPatchRequest;
+
+interface IdapiFetchOptions {
+	route: string;
+	method: string;
+	headers: {
+		'X-GU-ID-Client-Access-Token'?: string;
+		'X-GU-ID-FOWARDED-SC-GU-U'?: string;
```

Comment:
The client access token is the equivalent of an API key - it's for when a serverside app interacts with IDAPI, and usually that app can perform operations on any user.

The SC_GU_U cookie, which is being turned into a header here, is a per-user encrypted cookie which is set in their browser and gives that user access to specific endpoints in IDAPI which affect only that user. They're usually called `/user/me/something`, rather than the client access token protected endpoints which are called `/user/something`.

We're sending both headers in each request because there's no harm in doing so, I guess - ideally we'd only be sending one or the other, but I just lifted this setup from what we had before.

Once we've moved to OAuth, we won't be using either of these headers - the OAuth cookies will be scoped to specific operations instead.